swarm/network/discovery.go

```diff
@@ -160,12 +160,11 @@ func (d *Peer) handleSubPeersMsg(msg *subPeersMsg) error {
 	if !d.sentPeers {
 		d.setDepth(msg.Depth)
 		var peers []*BzzAddr
-		d.kad.EachConn(d.Over(), 255, func(p *Peer, po int) bool {
-			if pob, _ := Pof(d, d.kad.BaseAddr(), 0); pob > po {
-				return false
-			}
-			if !d.seen(p.BzzAddr) {
-				peers = append(peers, p.BzzAddr)
+		d.kad.EachConn(d.kad.BaseAddr(), 255, func(p *Peer, po int) bool {
```

Comment:
this is wrong!!! first arg isthe remote's overlay. advertised depth of the peer should be compared to the peer not  self. earlier was correct